File: 409-longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        if(s.size() == 1) return 1;
        unordered_map<char,int> counter;
        for(char c:s)counter[c]++;
        int count =0;
        bool flag = true;
        for(const auto& pair:counter){
            if(pair.second %2 != 0){
                count = count + pair.second -1;
                if(flag){
                    count++;
                    flag = false;
                }
            }
            else count = count + pair.second;
        }
        return count;

    }
};

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        if len(s) == 1:
            return 1
        ans = Counter(s)
        flag = True
        count = 0
        for i in ans.keys():
            if ans[i] %2 != 0:
                count+=ans[i] - 1
                if flag:
                    count+=1
                    flag = False
            else:
                count += ans[i]

        return count